Bluetooth: controller: Fix tx power level set and get

cvinayak opened the pull request:
Fix implementation to correctly return the configured
default Tx Power Level.

Also, fix the missing use of RADIO_TXP_DEFAULT in the new
ULL/LLL implementation of Advertiser and Observer states.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: controller: Fix tx power level set and get

Fix implementation to correctly return the configured
default Tx Power Level.

Also, fix the missing use of RADIO_TXP_DEFAULT in the new
ULL/LLL implementation of Advertiser and Observer states.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

SebastianBoe commented:
Fix for one of (all of?) the CI failures: https://github.com/zephyrproject-rtos/zephyr/commit/6cc787061ea6394e82273c658e243c88355934ab

cvinayak committed:
Bluetooth: controller: Read Tx Power Level depends on Connections

Conditionally compile HCI Read Tx Power Level command for
connections support.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>
Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

SebastianBoe commented:
Hi, could you please explain how reading the TX power level depends on connections?

Intuitively, I would think you could check the TX power level independent for beacons as well.

cvinayak commented:
@SebastianBoe Its is HCI Spec requirement on the interface to read Tx power used by controller for a connection. And, a controller implementation may use adaptive Tx power based on measured peer Rx RSSI (which is not in our controller), in these cases a host will query Tx power of each connection by handle.
